index.js

```diff
@@ -284,74 +324,112 @@ class GithubScm extends Scm {
+                return resolve({
+                    avatar: data.avatar_url,
+                    name: data.name,
+                    username: data.login,
+                    url: data.html_url
+                });
             });
         });
     }
 
     /**
-     * Get repository object for storing new pipeline
-     * @method _getRepoId
-     * @param  {Object}    config        Configuration
-     * @param  {String}    config.scmUrl The scmUrl to get permissions on
-     * @param  {String}    config.token  The token used to authenticate to the SCM
+     * Decorate the commit based on the repository
+     * @method _decorateCommit
+     * @param  {Object}        config        Configuration object
+     * @param  {Object}        config.scmUri SCM URI the commit belongs to
+     * @param  {Object}        config.sha    SHA to decorate data with
+     * @param  {Object}        config.token  Service token to authenticate with Github
      * @return {Promise}
      */
-    _getRepoId(config) {
-        const scmInfo = getInfo(config.scmUrl);
+    _decorateCommit(config) {
+        const scmLookup = this.lookupScmUri({
+            scmUri: config.scmUri,
+            token: config.token
+        });
+        const commitLookup = scmLookup.then(scmInfo =>
+            new Promise((resolve, reject) => {
+                this.breaker.runCommand({
+                    action: 'getCommit',
+                    token: config.token,
+                    params: {
+                        owner: scmInfo.user,
+                        repo: scmInfo.repo,
+                        sha: config.sha
+                    }
+                }, (error, data) => {
+                    if (error) {
+                        return reject(error);
+                    }
+
+                    return resolve(data);
+                });
+            })
+        );
+        const authorLookup = commitLookup.then(commitData =>
+            this.decorateAuthor({
+                token: config.token,
+                username: commitData.committer.login
+            })
+        );
 
         return Promise.all([
-            // eslint-disable-next-line no-underscore-dangle
-            this._getRepoInfo(scmInfo, config.token),
-            // eslint-disable-next-line no-underscore-dangle
-            this._getBranchUrl(scmInfo, config.token)
-        ]).then(([repoInfo, branchUrl]) => ({
-            id: `${scmInfo.host}:${repoInfo.id}:${scmInfo.branch}`,
-            name: repoInfo.full_name,
-            url: branchUrl
-        }));
+            scmLookup,
+            commitLookup,
+            authorLookup
+        ]).then(([scmData, commitData, authorData]) => {
+            const urlBase = `https://${scmData.host}/${scmData.user}/${scmData.repo}`;
+
+            return {
+                author: authorData,
+                message: commitData.commit.message,
+                url: `${urlBase}/tree/${config.sha}`
```

Comment:
We can get this directly from `commitData.html_url`. Example payload: https://api.github.com/repos/screwdriver-cd/data-schema/commits/0d9649e385fcfdb282a33eada3757f9b186c1289

The only difference is `/commits/` instead of `/tree/`. Talked to @minz1027, we think the `commits` might be more useful for the UI since it shows the difference